mlkem/native/x86_64/src/basemul.S

```diff
@@ -52,14 +50,20 @@ vmovdqa		%ymm13,(%rsp)
 vpmullw		%ymm5,%ymm9,%ymm13			# a0c0.lo
 vpmullw		%ymm6,%ymm9,%ymm9			# a0d0.lo
 vpmullw		%ymm5,%ymm10,%ymm5			# b0c0.lo
-vpmullw		%ymm6,%ymm10,%ymm10			# b0d0.lo
 
 /* Compute low-parts of monomials in (a1+b1*X)*(c1+d1*X), */
 /* using Montgomery twists calculated before */
 vpmullw		%ymm7,%ymm11,%ymm6			# a1c1.lo
 vpmullw		%ymm8,%ymm11,%ymm11			# a1d1.lo
 vpmullw		%ymm7,%ymm12,%ymm7			# b1c1.lo
-vpmullw		%ymm8,%ymm12,%ymm12			# b1d1.lo
+
+/* Use cached (d * zeta) to compute the high part of (b*d) terms */
+vmovdqa		(32*\off+ 0)*2(%r8),%ymm8	# d0z
```

Comment:
Now that we use the mulcache, can we accumulate in 32bit and reduce once in the end? That's how it's done in C and AArch64 backend, and the mulcache enables it (otherwise there are two chained muls anyway).

Reply:
I thought about this before opening the PR and I don't think that can be easily done with AVX2 because if you want to accumulate in 32bit then the multiplication also has to be in 32bit, so we'll basically have to load the coefficients into ymm's as 32-bit instead of 16-bit integers and work like that. It's definitely worth exploring so that's going to be my next mini project :)

Reply:
Can one not load in 16bit, use separate hi/login bit multiply, zip to 32bit, accumulate in 32bit, and reduced? It certainly has more Overhead than AArch64 where one has a widening multiply to begin with. But perhaps pairs of lo/hi multiply are anyway fused? I don't know Intel

Reply:
yes, but that's the same as I described with the order of operations a bit different. My suggestion is to load the coefficients in 16bit and then zip to 32bit with a zero vector, then single mul and reduce. IIUC you are suggesting to change the order of mul and zip (and do mullo/mulhi). If we do it my way then we don't have to worry about pairs of mul lo/hi being fused or not.

Reply:
Yes but a 32bit mul may be more expensive than 2x16bit muls. We'll see!

Reply:
I don't think so, according to https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html both have latency of 5 cycles and throughput of 0.5 CPI on Intel architectures. Anyway, I can test both and see which one is better performing. But I'd rather do that in a new PR if that's ok with you?

Reply:
Yes I think that's better - this PR is already an improvement and a good first step, and soon ready to go once we get past the tooling issues.